src/benchmarks.zig

```diff
@@ -1,349 +1,538 @@
-    mean_runtime: anytype,
-) ![6]u64 {
-    const arg_len = std.fmt.count("{}", .{arg_name});
-    const name_len = try alignedPrint(writer, .left, min_widths[0], "{s}{s}{}{s}", .{
-        func_name,
-        "("[0..@intFromBool(arg_len != 0)],
-        arg_name,
-        ")"[0..@intFromBool(arg_len != 0)],
-    });
-    try writer.writeAll(" ");
-    const it_len = try alignedPrint(writer, .right, min_widths[1], "{}", .{iterations});
-    try writer.writeAll(" ");
-    const min_runtime_len = try alignedPrint(writer, .right, min_widths[2], "{}", .{min_runtime});
-    try writer.writeAll(" ");
-    const max_runtime_len = try alignedPrint(writer, .right, min_widths[3], "{}", .{max_runtime});
-    try writer.writeAll(" ");
-    const variance_len = try alignedPrint(writer, .right, min_widths[4], "{}", .{variance});
-    try writer.writeAll(" ");
-    const mean_runtime_len = try alignedPrint(writer, .right, min_widths[5], "{}", .{mean_runtime});
-
-    return [_]u64{ name_len, it_len, min_runtime_len, max_runtime_len, variance_len, mean_runtime_len };
-}
+    // print the results in a formatted table
+    inline for (functions, 0..) |def, fcni| {
+        var fmt_buf: [512]u8 = undefined;
+        const file_name_average = try std.fmt.bufPrint(&fmt_buf, "{s}/{s}.csv", .{ benchmark_name, def.name });
+        const file_average = try results_dir.openFile(file_name_average, .{});
+        defer file_average.close();
+
+        var table = pt.Table.init(allocator);
+        defer table.deinit();
+        var read_buf: [1024 * 1024]u8 = undefined;
+        try table.readFrom(file_average.reader(), &read_buf, ",", true);
+
+        if (!is_multi_return.items[fcni]) {
+            // direct print works ok in this case
+            try table.printstd();
+        } else {
+            // re-parse the return type
+            const benchFunction = @field(B, def.name);
+            const arguments = blk: {
+                switch (@typeInfo(@TypeOf(benchFunction))) {
+                    .Fn => |info| {
+                        // NOTE: to know if we should pass in the time unit we
+                        // check the input params of the function, so any multi-return
+                        // function NEEDS to have the time unit as the first parameter
+                        if (info.params.len > 0 and info.params[0].type.? == BenchTimeUnit) {
+                            break :blk switch (@TypeOf(args[0])) {
+                                void => .{time_unit},
+                                else => .{ time_unit, args[0] },
+                            };
+                        } else {
+                            // single return type
+                            break :blk switch (@TypeOf(args[0])) {
+                                void => .{},
+                                else => .{args[0]},
+                            };
+                        }
+                    },
+                    else => unreachable,
+                }
```

Comment:
Same kind of suggestion as on other `switch (@typeInfo(@TypeOf(benchFunction)))`

Reply:
8948a0cb3b14df8d6d5d65356aefa394a1c27ee8